spec/domain_viewer_tracker_spec.rb

```diff
@@ -1,11 +1,125 @@
+    TestModel.new
   end
+  let(:test_cookie) { ActionDispatch::Cookies::CookieJar.new(nil) }
 
-  it 'does something useful' do
-    expect(false).to eq(true)
+  describe "#set_viewer_id" do
+    subject { testModel.send(:set_viewer_id) }
+
+    let(:testModel) do
+      class TestModel
+        include DomainViewerTracker
+      end
+      allow_any_instance_of(TestModel).to receive(:cookies).and_return(test_cookie)
+      TestModel.new
+    end
+    let(:test_cookie) { ActionDispatch::Cookies::CookieJar.new(nil) }
+
+    it "should save cookie with uuid" do
+      subject
+      expect(test_cookie[:viewer_id]).to match /^.{36}$/
+    end
+
+    context "key name is changed" do
+      before do
+        DomainViewerTracker.configure do |config|
+          config.cookie_key_name = "sushi"
+        end
+      end
+
+      it "should save cookie with changed key" do
+        subject
+        expect(test_cookie[:viewer_id]).to be_nil
+        expect(test_cookie[:sushi]).to match /^.{36}$/
+      end
+
+      after do
+        DomainViewerTracker.configure do |config|
+          config.cookie_key_name = "viewer_id"
+        end
+      end
+    end
+
+    context "domain is changed" do
+      before do
+        DomainViewerTracker.configure do |config|
+          config.cookie_domain = ".sushi.example.com"
+        end
+      end
+
+      it "should save cookie with changed key" do
+        subject
+        expect(test_cookie[:viewer_id]).to match /^.{36}$/
+        expect(test_cookie.instance_eval("@set_cookies")["viewer_id"][:domain]).to eq ".sushi.example.com"
+      end
+
+      after do
+        DomainViewerTracker.configure do |config|
+          config.cookie_domain = ""
+        end
+      end
+    end
+  end
+
+  describe "#store_viewer_id" do
+    subject { testModel.send(:store_viewer_id, user_id) }
+
+    context "cookie and user_id exist" do
+      let(:user_id) { 1 }
+      before do
+        test_cookie[:viewer_id] = "hoge"
+      end
+
+      it "Creates viewer record." do
+        expect { subject }.to change { Viewer.count }.by 1
+        expect(Viewer.exists?(uuid: "hoge", user_id: user_id)).to be true
+      end
+    end
+
+    context "cookie is missing" do
+      let(:user_id) { 1 }
+
+      it "Creates viewer record." do
+        expect { subject }.to change { Viewer.count }.by 1
+        expect(Viewer.exists?(uuid: nil, user_id: user_id)).to be true
+      end
+    end
+
+    context "user_id is missing" do
+      let(:user_id) { nil }
+
+      before do
+        test_cookie[:viewer_id] = "hoge"
+      end
+
+      it "Creates viewer record." do
+        expect { subject }.to change { Viewer.count }.by 1
+        expect(Viewer.exists?(uuid: "hoge", user_id: nil)).to be true
+      end
+    end
+
+    context "Viewer record already exist" do
+      let(:user_id) { 1 }
+
+      before do
+        test_cookie[:viewer_id] = "hoge"
+        Viewer.create!(uuid: "hoge", user_id: 1)
+      end
+
+      it "Creates viewer record." do
```

Comment:
I think the description is not correct.
I think "Does not creates viewer record" is correct.
What do you think?